dora/src/semck/globaldef.rs

```diff
@@ -390,6 +417,40 @@ mod tests {
     use crate::error::msg::SemError;
     use crate::semck::tests::*;
 
+    #[test]
+    fn test_class() {
+        err(
+            "class Foo class Foo",
+            pos(1, 11),
+            SemError::ShadowClass("Foo".into()),
+        );
+        err(
+            "fun Foo() {} class Foo",
+            pos(1, 14),
+            SemError::ShadowFunction("Foo".into()),
+        );
+        err(
+            "class Foo fun Foo() {}",
+            pos(1, 11),
+            SemError::ShadowClassConstructor("Foo".into()),
+        );
+        err(
+            "class Foo let Foo: Int = 1;",
```

Comment:
Could you please add a test case for a local variable as well?

Reply:
You mean like

```
class Foo {
    var Foo: Foo = nil;
}
```

or something different?

Reply:
@dinfuehr I added some checks here: https://github.com/dinfuehr/dora/pull/192/files#diff-313b446e400b66c47affefe1a6b19787R619

Reply:
Yes this PR is almost good to go. I think we should have a test for clashes between class name and local variable as well: "class Foo fun x() { let Foo: Int = 1; }". AFAIU this is allowed, but I think it would to add an explicit test for that.

Reply:
I'm on it! Give me 5 minutes...

Reply:
It wasn't allowed – I added a commit that allows it: https://github.com/dinfuehr/dora/pull/192/commits/25671411cd63c563d871c3bf0e9bc8c31a439dd9

Reply:
If you are able to merge this, how long are you available after that? If you are still here for a while, I'd rebase the other PRs on the merged one and try to get them through, too.